### PR TITLE
Cleanup dependencies

### PR DESCRIPTION
**Pull request by jsvd**

### CODE REVIEW

File: Gemfile.template [hunk 1]
```diff
@@ -5,27 +5,22 @@ source "https://rubygems.org"
 gem "logstash-core", :path => "./logstash-core"
 gem "logstash-core-plugin-api", :path => "./logstash-core-plugin-api"
 gem "paquet", "~> 0.2.0"
-gem "ruby-progressbar", "~> 1.8.1"
-gem "builder", "~> 3.2.2"
-gem "ci_reporter_rspec", "1.0.0", :group => :development
-gem "tins", "1.6", :group => :development
+gem "ruby-progressbar", "~> 1"
```
Comment (jsvd) [inline Gemfile.template]:
```suggestion
gem "ci_reporter_rspec", "1.0.0", :group => :development
gem "ruby-progressbar", "~> 1"
```

File: Gemfile.template [hunk 2]
```diff
@@ -5,27 +5,23 @@ source "https://rubygems.org"
...
 gem "logstash-devutils", "= 1.3.5", :group => :development
 gem "benchmark-ips", :group => :development
-gem "octokit", "3.8.0", :group => :build
+gem "octokit", "~> 4", :group => :build
 gem "stud", "~> 0.0.22", :group => :build
-gem "fpm", "~> 1.3.3", :group => :build
-gem "rubyzip", "~> 1.2.1", :group => :build
-gem "gems", "~> 0.8.3", :group => :build
-gem "rack-test", :require => "rack/test", :group => :development
+gem "fpm", "~> 1", :group => :build
```
Comment (jsvd) [inline Gemfile.template]:
```suggestion
gem "rack-test", :require => "rack/test", :group => :development
gem "fpm", "~> 1", :group => :build
```

File: Gemfile.template (hunk 1 above)
Comment (jsvd) [inline Gemfile.template]:
```suggestion
gem "ci_reporter_rspec", "1.0.0", :group => :development
gem "ruby-progressbar", "~> 1"
```

File: Gemfile.template (hunk 2 above)
Comment (jsvd) [inline Gemfile.template]:
```suggestion
gem "rack-test", :require => "rack/test", :group => :development
gem "fpm", "~> 1", :group => :build
```

File: logstash-core/logstash-core.gemspec [hunk 3]
```diff
@@ -47,17 +47,17 @@ Gem::Specification.new do |gem|
 
   gem.platform = "java"
 
-  gem.add_runtime_dependency "pry", "~> 0.10.1"  #(Ruby license)
+  gem.add_runtime_dependency "pry", "~> 0.12.2"  #(Ruby license)
```
Comment (yaauie) [inline logstash-core/logstash-core.gemspec]: pry as a runtime dependency is a bit surprising to me; are we sure it is needed at runtime and not just in test/development?

File: logstash-core/logstash-core.gemspec [hunk 4]
```diff
@@ -47,17 +47,17 @@ Gem::Specification.new do |gem|
...
   gem.add_runtime_dependency "stud", "~> 0.0.19" #(Apache 2.0 license)
   gem.add_runtime_dependency "clamp", "~> 0.6.5" #(MIT license) for command line args/flags
-  gem.add_runtime_dependency "filesize", "0.0.4" #(MIT license) for :bytes config validator
-  gem.add_runtime_dependency "gems", "~> 0.8.3"  #(MIT license)
-  gem.add_runtime_dependency "concurrent-ruby", "~> 1.0", ">= 1.0.5"
+  gem.add_runtime_dependency "filesize", "~> 0.2.0" #(MIT license) for :bytes config validator
+  gem.add_runtime_dependency "gems", "~> 1"  #(MIT license)
```
Comment (yaauie) [inline logstash-core/logstash-core.gemspec]: my reading of `~> 1` is functionally the same as `>= 1`, meaning if a 2.x is ever released (presumably with breaking changes of some sort), we will blindly consume it. Is this what we intend?

File: logstash-core/logstash-core.gemspec [hunk 5]
```diff
@@ -47,17 +47,17 @@ Gem::Specification.new do |gem|
...
+  gem.add_runtime_dependency "concurrent-ruby", "~> 1"
   gem.add_runtime_dependency "rack", '~> 1.6', '>= 1.6.11'
-  gem.add_runtime_dependency "sinatra", '~> 1.4', '>= 1.4.6'
-  gem.add_runtime_dependency 'puma', '~> 2.16'
-  gem.add_runtime_dependency "jruby-openssl", ">= 0.9.20" # >= 0.9.13 Required to support TLSv1.2
-  gem.add_runtime_dependency "chronic_duration", "0.10.6"
+  gem.add_runtime_dependency "sinatra", '~> 1', '>= 1.4.6'
+  gem.add_runtime_dependency 'puma', '~> 2'
```
Comment (yaauie) [inline logstash-core/logstash-core.gemspec]: I'm also wary of the `~> 2` here; if puma ever releases a 3.x, we'll silently upgrade which could be hard to debug.

**Comment by jsvd** [comment]
> Since the squiggle-arrow operator allows the most-precisely specified version field to be greater than or equal to what is specified, using it with a single version field (the major) presumably allows breaking changes when dependencies release new majors; this could produce hard-to-debug problems later on if/when new major versions are released.

The squiggle operator is a bit weird for majors, it behaves as "only allow this major". This can be demonstrated with:

```
jruby-9.2.3.0 :007 > Gem::Requirement.new("~> 1").satisfied_by?(Gem::Version.new("2.0.0"))
 => false 
jruby-9.2.3.0 :008 > Gem::Requirement.new("~> 1").satisfied_by?(Gem::Version.new("2"))
 => false 
jruby-9.2.3.0 :009 > Gem::Requirement.new("~> 1").satisfied_by?(Gem::Version.new("1.4"))
 => true 
jruby-9.2.3.0 :010 > Gem::Requirement.new("~> 1").satisfied_by?(Gem::Version.new("0.4"))
 => false 
jruby-9.2.3.0 :011 > Gem::Requirement.new("~> 1").satisfied_by?(Gem::Version.new("1.0"))
 => true
```

**Comment by jsvd** [comment]
jenkins test this please

File: logstash-core/logstash-core.gemspec (hunk 3 above)
Comment (jsvd) [reply logstash-core/logstash-core.gemspec]: pry in runtime is useful because we support `bin/logstash -i irb` and `bin/logstash -i pry`

**Comment by jsvd** [comment]
@yaauie ci is green now, can you take another pass?

**Pull request merged**